src/gt4py/cartesian/backend/pyext_builder.py

```diff
@@ -112,9 +113,9 @@ def get_gt_pyext_build_opts(
             "--compiler-options",
             "-fPIC",
         ]
-    extra_link_args = gt_config.build_settings["extra_link_args"]
+    extra_link_args = copy.deepcopy(gt_config.build_settings["extra_link_args"])
 
-    mode_flags = ["-O0", "-ggdb"] if debug_mode else ["-O3", "-DNDEBUG"]
+    mode_flags = [f"-O{opt_level}", "-DNDEBUG", *extra_opt_flags.split()]
```

Comment:
@havogt On (my) mac, the flag `-DNDEBUG` is required to make compilation works with any optimization level, so I hardcoded it into `mode_flags` for the sake of convenience. If you prefer, we could be more flexible: 
```suggestion
    mode_flags = [f"-O{opt_level}"]
    if extra_opt_flags != "":
        mode_flags += [*extra_opt_flags.split()]
    elif opt_level == "0":
        mode_flags.append("-ggdb")
    else:
        mode_flags.append("-DNDEBUG")
```